File: legacy_code/strings/alternating_chars.py
#!/bin/python

def count_repeat(s):
    if len(s) == 0:
        return 0
    current = s[0]
    repeat = 0
    for i in xrange(1, len(s)):
        if current == s[i]:
            repeat += 1
        current = s[i]
    return repeat

t = int(raw_input())
for _ in xrange(t):
    s = raw_input()
    print count_repeat(s)

File: legacy_code/strings/alternating_chars.cpp
#include <iostream>

using namespace std;

int count_repeat(string s) {
    if (s.length() == 0) return 0;
    char current = s.at(0);
    int repeat = 0;
    for (int i = 1; i < s.length(); i++) {
        char new_char = s.at(i);
        if (current == new_char) repeat++;
        current = new_char;
    }
    return repeat;
}

int main() {
    int t;
    cin >> t;
    for (int i = 0; i < t; i++) {
        string s;
        cin >> s;
        cout << count_repeat(s) << endl;
    }
    return 0;
}
